Add pixi to unit-test hook

If you try to make a commit outside of a pixi shell, you get this:

```
unit-tests...............................................................Failed
- hook id: unit-tests
- exit code: 4

ImportError while loading conftest '/Users/katie/Documents/work/catalyst/pudl-archiver/tests/conftest.py'.
tests/conftest.py:7: in <module>
    from pudl_archiver.frictionless import DataPackage
E   ModuleNotFoundError: No module named 'pudl_archiver'
```

We should see if there's a way to make the hooks work regardless of whether you're developing with `pixi run` or `pixi shell`

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -69,7 +69,7 @@ repos:
         verbose: false
         pass_filenames: false
         always_run: true
-        entry: pytest tests/unit
+        entry: pixi run pytest tests/unit
 
 # Configuration for pre-commit.ci
 ci:

